CHECKLIST.md

```diff
@@ -1611,5 +1611,38 @@ _Details coming soon…_
 
 _Details coming soon…_
 
+## 3. Method Design Checklist
+
+### 3.6.12. **Consider** a builder when faced with many constructor parameters
+
+#### Rationale:
+
+Telescoping constructors are hard to use to read and maintain. They are also error-prone, as it's easy to skip some
+combinations. The problem grows with the list of parameters, when we need to create a constructor for each combination.
+
+A builder can solve this problem.
+
+##### Builder
+
+Pros:
+ 
+ - no need for deprecation if we add a new required parameter
+ - fewer constructors to maintain
+ - less pain for the consumers, because they no longer have to pick between different constructor signatures
+ - automatic support for all subsets of parameters rather than arbitrary ones
+ - single source of defaults
+ - a way distinguish different arrangements of parameters of the same type
+ - a sustainable approach to copy objects
+ 
+Cons:
+ 
+ - a bit more code to write
+
+References:
+
+Bloch, J. (2018). Effective Java. Boston: Addison-Wesley. (Item 2).
+https://medium.com/@biratkirat/learning-effective-java-item-2-e06c9eee7d0
+https://ecosystem.atlassian.net/browse/JPERF-284
```

Comment:
These links don't render well. They're in raw form and next to each other in the same line.